pkg/benchmark/resources.go

```diff
@@ -0,0 +1,187 @@
+package benchmark
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type NodeResource struct {
+	NodeName          string   `json:"node_name"`
+	Disks             []string `json:"disks"`
+	NetworkInterfaces []string `json:"network_interfaces"`
+}
+
+type ClusterResources struct {
+	Nodes []NodeResource `json:"nodes"`
+}
+
+func CollectResources(outputPath string) error {
+	nodes, err := getClusterNodes()
+	if err != nil {
+		return err
+	}
+
+	var resources ClusterResources
+
+	for _, node := range nodes {
+		fmt.Printf("Collecting data for node: %s\n", node)
+		disks, err := getNodeDisks(node)
+		if err != nil {
+			return err
+		}
+		nics, err := getNodeNics(node)
+		if err != nil {
+			return err
+		}
+
+		resources.Nodes = append(resources.Nodes, NodeResource{
+			NodeName:          node,
+			Disks:             disks,
+			NetworkInterfaces: nics,
+		})
+	}
+
+	jsonBytes, err := json.MarshalIndent(resources, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(outputPath, jsonBytes, 0644)
+}
+
+func getClusterNodes() ([]string, error) {
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	if kubeconfigPath == "" {
+		kubeconfigPath = os.ExpandEnv("$HOME/.kube/config")
+	}
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load kubeconfig: %v", err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create clientset: %v", err)
+	}
```

Comment:
```suggestion
	kubeconfigPath := os.Getenv("KUBECONFIG")
	if kubeconfigPath == "" {
		kubeconfigPath = os.ExpandEnv("$HOME/.kube/config")
	}

	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
	if err != nil {
		return nil, fmt.Errorf("failed to load kubeconfig: %v", err)
	}
	clientset, err := kubernetes.NewForConfig(config)
	if err != nil {
		return nil, fmt.Errorf("failed to create clientset: %v", err)
	}
```

You don't need this, we can get these details by calling `getclientsets` in the cmd folders